packages/valory/skills/abstract_round_abci/tests/test_tools/test_rounds.py

```diff
@@ -0,0 +1,689 @@
+        if self.collection_threshold_reached and self.dummy_exit_event is not None:
+            return (
+                cast(
+                    DummySynchronizedData,
+                    self.synchronized_data.update(
+                        most_voted_keeper_address=list(self.collection.keys())
+                    ),
+                ),
+                self.dummy_exit_event,
+            )
+        return None
+
+
+class TestBaseCollectDifferentUntilAllRoundTest(BaseTestBase):
+    """Test `BaseCollectDifferentUntilAllRoundTest`."""
+
+    base_round_test: BaseCollectDifferentUntilAllRoundTest
+    base_round_test_cls: Type[
+        BaseCollectDifferentUntilAllRoundTest
+    ] = BaseCollectDifferentUntilAllRoundTest
+
+    @given(
+        st.one_of(st.none(), st.sampled_from(DummyEvent)),
+    )
+    def test_test_round(self, exit_event: DummyEvent) -> None:
+        """Test `_test_round`."""
+        self.base_round_test.consensus_params._max_participants = (  # pylint: disable=protected-access
+            MAX_PARTICIPANTS
+        )
+        test_round = DummyCollectDifferentUntilAllRoundWithEndBlock(
+            exit_event,
+            self.base_round_test.synchronized_data,
+            self.base_round_test.consensus_params,
+        )
+        round_payloads = [
+            (DummyTxPayload(f"agent_{i}", i)) for i in range(MAX_PARTICIPANTS)
+        ]
+        synchronized_data_attr_checks = [
+            lambda _synchronized_data: _synchronized_data.most_voted_keeper_address
+        ]
+
+        # Create `_test_round` generator and trigger for the first time.
+        gen = self.base_round_test._test_round(  # pylint: disable=protected-access
+            test_round,
+            cast(List[BaseTxPayload], round_payloads),
+            lambda synchronized_data, _: synchronized_data.update(
+                most_voted_keeper_address=[
+                    f"agent_{i}" for i in range(MAX_PARTICIPANTS)
+                ]
+            ),
+            synchronized_data_attr_checks,
+            exit_event,
+        )
+        for _ in range(4):
+            next(gen)
+        last_iteration(gen)
+
+
+class DummyCollectSameUntilAllRoundWithEndBlock(DummyCollectSameUntilAllRound):
+    """A `DummyCollectSameUntilAllRound` with `end_block` implemented."""
+
+    def __init__(self, dummy_exit_event: DummyEvent, *args: Any, **kwargs: Any):
+        """Initialize the dummy class."""
+        super().__init__(*args, **kwargs)
+        self.dummy_exit_event = dummy_exit_event
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
+        """A dummy `end_block` implementation."""
+        if self.collection_threshold_reached:
+            return (
+                cast(
+                    DummySynchronizedData,
+                    self.synchronized_data.update(
+                        most_voted_keeper_address=self.common_payload
+                    ),
+                ),
```

Comment:
can we introduce a helper function (with `**kwags` signature) for this on the base class for tests at least? it's a repeated all over the place (6x) and will make the code much more readable.